lib/decorators/api-extension.decorator.ts

```diff
@@ -9,10 +9,7 @@ export function ApiExtension(extensionKey: string, extensionProperties: any) {
   }
 
   const extensionObject = {
-    [extensionKey]:
-      typeof extensionProperties !== 'string'
-        ? { ...extensionProperties }
-        : extensionProperties
+    [extensionKey]: extensionProperties
```

Comment:
This condition was there for a reason (I'm not sure but I believe the reason was that we wanted to make sure the original `extensionProperties` object won't get mutated and so we clone it). Instead of removing it, I'd suggest adding another "branch" of this condition, making sure we don't spread the object when it's an array (but run sth like `[].concat(extenstionProperties)` or `[...extenstionProperties]`)

Reply:
This can work:

```suggestion
    [extensionKey]: clone(extensionProperties)
```

(with `clone` imported from `lodash`)